src/main/java/com.company/tree/TopView.java

```diff
@@ -0,0 +1,26 @@
+package com.company.tree;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+public class TopView {
+    List<Integer> topView(TreeNode root) {
+        TreeMap<Integer, Integer> map = new TreeMap<>();
+        dfs(root, 0, map);
+        return new ArrayList<Integer>(map.values());
+    }
+
+    private void dfs(TreeNode node, int horizontalDistance, TreeMap map) {
```

Comment:
Shouldn't it be vertical? It's top view

Reply:
in vertical order we print all elements in each vertical column

in top view we print only top element in each vertical column